File: Backend/TaxManagementSystem/src/main/java/com/tms/Maker/entity/TaxableSearchEngine.java
package com.tms.Maker.entity;

import java.time.LocalDate;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class TaxableSearchEngine {

    private Long branch_id;
    private Long tax_category_id;
    private String reference_number;
    private String router_status;

    private List<LocalDate> maked_date;
    private List<LocalDate> checked_date;
    private List<LocalDate> approved_date;
    private List<LocalDate> rejected_date;

    private String document_type;
    private Long user_id;
    private String search_by;
    private Long director_id;

    // pagination
    private int totalRecords;
    private int currentPage;
    private int pageSize;
    private int event_length;
}

File: Frontend/src/app/models/common/taxable-search-engine.ts
export class TaxableSearchEngine {
    branch_id?:number;
    tax_category_id?:number;
    reference_number?:number;
    router_status?: string;
    maked_date?: Date[];
    checked_date?: Date[];
    approved_date?: Date[];
    rejected_date?: Date[];
    document_type?: string;
    user_id?: number;
    search_by?: string; 
    director_id?: number;

    // For Paginatoruser_id?: number;
    totalRecords: number = 0;
    currentPage: number = 0; 
    pageSize: number = 5;
    event_length: number = 0;
}